apps/platform/src/app/(main)/(project)/[workspace]/[project]/@secret/page.tsx

```diff
@@ -47,13 +48,15 @@ function SecretPage(): React.JSX.Element {
   const [hasMore, setHasMore] = useState(true)
   const [isLoading, setIsLoading] = useState<boolean>(true)
 
-  const privateKey = useMemo(
-    () =>
-      selectedProject?.storePrivateKey
-        ? selectedProject.privateKey
-        : localStorage.getItem(`${selectedProject?.name}_pk`) || null,
-    [selectedProject]
-  )
+  const [privateKey, setPrivateKey] = useAtom(selectedProjectPrivateKeyAtom)
+
+  useEffect(() => {
+    const key = selectedProject?.storePrivateKey
+      ? selectedProject.privateKey
+      : localStorage.getItem(`${selectedProject?.slug}_pk`) || null
+    setPrivateKey(key)
+  }, [selectedProject, setPrivateKey])
```

Comment:
since we are using this logic more than once, I think we should make a hook for this with the name `useFetchPrivateKey` or `useFetchProjectPrivateKey`, and while working on the hook pls refactor the logic to
```tsx
useEffect(() => {
  if (!selectedProject) {
    setIsLoading(false)
    return
  }
  
  const fetchPrivateKey = () => {
    // From server if stored there
    if (selectedProject.storePrivateKey) {
      return selectedProject.privateKey
    }
    
    // Otherwise try local storage
    const localStorageKey = `${selectedProject.slug}_pk`
    return localStorage.getItem(localStorageKey) || null
  }

  setPrivateKey(fetchPrivateKey())
  setIsLoading(false)
}, [selectedProject, setPrivateKey])
```